### PR TITLE
feat(tests): EIP-7928 OOG edge cases for EIP-7702 delegations

### PR DESCRIPTION
**Pull request by raxhvl**
## 🗒️ Description

The PR adds two test cases targeting OOG scenarios when accessing 7702 delegation:

1. Case 1: OOG at Authority access; Neither authority or delegation target is expected in BAL.
2. Case 2: OOG at Delegation access; Authority MUST be present but Delegation target is NOT expected in BAL.

The overall goal is to ensure when an account access runs out of gas, it does not make it to BAL.

## 🔗 Related Issues or PRs
N/A.

## ✅ Checklist

- [ ] All: Ran fast `tox` checks to avoid unnecessary CI fails, see also [Code Standards](https://eest.ethereum.org/main/getting_started/code_standards/) and [Enabling Pre-commit Checks](https://eest.ethereum.org/main/dev/precommit/):
    ```console
    uvx tox -e static
    ```
- [x] All: PR title adheres to the [repo standard](https://eest.ethereum.org/main/getting_started/contributing/?h=contri#commit-messages-issue-and-pr-titles) - it will be used as the squash commit message and should start `type(scope):`.
- [ ] All: Considered adding an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [ ] All: Set appropriate labels for the changes (only maintainers can apply labels).

#### Cute Animal Picture
```sh
    |
    \\ A _.-''.-""
    /`''`    /
   |__  @)   |
    '-....--'\
              \
              |
```

### CODE REVIEW

File: tests/amsterdam/eip7928_block_level_access_lists/test_block_access_lists_eip7702.py [hunk 1]
```diff
@@ -828,3 +830,170 @@ def test_bal_7702_double_auth_swap(
...
+) -> None:
+    """Ensure BAL handles OOG during EIP-7702 authority account access."""
+    alice = pre.fund_eoa()
+    target_contract = pre.deploy_contract(code=Op.STOP)
+    gas_costs = fork.gas_costs()
+
+    # Bob is a pre-deployed account with 7702 delegation to target_contract
+    bob = pre.deploy_contract(
+        nonce=1,
+        code=Spec7702.delegation_designation(target_contract),
```
Comment (jochem-brouwer) [inline tests/amsterdam/eip7928_block_level_access_lists/test_block_access_lists_eip7702.py:1091]: I'm not sure how the test framework translates this, but if it tries to deploy a contract with the designated 7702 delegation pointer then this will not work (contract deployment will fail because it attempts to deploy 0xEF-prefixed code (see EIP-3541)

File: tests/amsterdam/eip7928_block_level_access_lists/test_block_access_lists_eip7702.py [hunk 2]
```diff
@@ -828,3 +830,170 @@ def test_bal_7702_double_auth_swap(
...
+    #
+    # We provide enough gas for Bob access but OOG before TargetContract access
+    push_cost = gas_costs.G_VERY_LOW * 7
+    bob_cold_cost = gas_costs.G_COLD_ACCOUNT_ACCESS
+    target_cold_cost = gas_costs.G_COLD_ACCOUNT_ACCESS
+
+    # Provide enough for intrinsic + pushes + Bob access,
+    # but not TargetContract
+    tx_gas_limit = (
+        intrinsic_gas_cost + push_cost + bob_cold_cost + target_cold_cost - 1
```
Comment (jochem-brouwer) [inline tests/amsterdam/eip7928_block_level_access_lists/test_block_access_lists_eip7702.py:1210]: This test is almost similar to the previous one, right? The only difference is the gas limit of the tx and the expected BAL 🤔 I think those should be merged in one test (and it should be parametrized)

**Review by jochem-brouwer — COMMENTED** [review]
Two comments 😄 👍

File: tests/amsterdam/eip7928_block_level_access_lists/test_block_access_lists_eip7702.py (hunk 1 above)
Comment (raxhvl) [reply tests/amsterdam/eip7928_block_level_access_lists/test_block_access_lists_eip7702.py:1091]: The deployment goes through (we've done the same for the other tests). My understanding that this just adds to the pre-state before invoking EVM so its fine. 


cc: @fselmo

Comment (jochem-brouwer) [reply tests/amsterdam/eip7928_block_level_access_lists/test_block_access_lists_eip7702.py:1091]: Ah right, yes this would work to generate the fixtures but it would fail on a live network (`uv run execute remote`)

**Comment by fselmo** [comment]
@raxhvl, we had a lot of changes this time that came from `forks/osaka` and I rebased here. Just letting you know :)

**Comment by fselmo** [comment]
I believe all these are covered by PR #1882 that was merged. Can you confirm @raxhvl? Closing for now
